src/update/update.go

```diff
@@ -68,10 +69,9 @@ func CheckAndUpdate(config *core.Configuration, updatesEnabled, updateCommand, f
 		fmt.Fprintf(os.Stderr, "%s Please from version %s to %s", word, pleaseVersion(), config.Please.Version.VersionString())
 	}
 
-	// Must lock here so that the update process doesn't race when running two instances
-	// simultaneously.
-	core.AcquireRepoLock()
-	defer core.ReleaseRepoLock()
+	// Must lock exclusively here so that the update process doesn't race when running two instances simultaneously.
+	// We don't want to release the lock yet since other parts of the code might override it as needed.
```

Comment:
I'm not sure I follow this. I would expect `AcquireExclusiveRepoLock` to always be paired with a deferred release?

Reply:
Actually, in other for this to be more correct, I needed to `defer AcquireSharedRepoLock()`. This was working before because `AcquireSharedRepoLock` is called in `runPlease`. Since the repo lock fd is a singleton, replacing the lock mode is enough.